specs/language/resources.tex

```diff
@@ -0,0 +1,952 @@
+\Ch{Resource Types}{Resources}
+
+Resources are built-in intangible types representing memory with an external interface.
+
+These take the form of Typed Buffers, Raw Buffers, Textures, and Samplers.
+Buffer and Texture types can be read-only or writable.
+
+\Sec{Typed Buffers}{Resources.tybuf}
+
+The typed buffer class template represents a one-dimensional resource containing an array of a single given type.
+Its contents are indexed by typed access to each element
+Types may have their formats converted upon load.
+
+Template types can be any type that totals or can be padded to 16 bytes.
+
+All typed buffers can be read through subscript operators or Load methods.
+Writable typed buffers can be written through subscript operators.
+
+Where \texttt{T} can be any arithmetic type \ref{Basic.types.arithmetic}
+or a vector with a maximum of four elements containing such an arithmetic type.
+The total size of a single contained element must be less than 16 bytes.
+
+Typed buffers perform format conversions on load such that the underlying data
+gets converted to the destination type.
+
+\Sec{Read-only Typed Buffers}{Resources.rotybufs}
+
+\begin{HLSL}
+template <typename T = float4>
+ class Buffer {
+   Buffer();
```

Comment:
We also need a copy constructor.

Reply:
We use references elsewhere and when we do add references to the language I think it'll mean that this documentation just becomes correct.  

I think that the copy constructors should take const references though.